src/index.js

```diff
@@ -1,24 +1,29 @@
 import Storage from './storage.js';
 import Names from './names.js';
 import Messages from './messages.js';
-import Contract from './contracts.js';
+import Blockchain from './blockchain.js';
 import Utils from './utils.js';
 
 var EmbarkJS = {
   onReady: function (cb) {
-    if (typeof (__embarkContext) === 'undefined') {
+    if (self.Blockchain.done) {
       return cb();
     }
-    return __embarkContext.execWhenReady(cb);
+    this.Blockchain.finalCb = function() {
+      cb();
+    }
   }
 };
 
-EmbarkJS.Contract = Contract;
+EmbarkJS.Blockchain = Blockchain;
 EmbarkJS.Storage = Storage;
 EmbarkJS.Names = Names;
 EmbarkJS.Messages = Messages;
 EmbarkJS.Utils = Utils;
+EmbarkJS.Contract = function() {
+  console.error("deprecated: please use EmbarkJS.Blockchain.Contract instead");
```

Comment:
Should probably throw an error instead. Because it will throw anyway at some place else